### PR TITLE
Add hostif trap for NTP

### PR DESCRIPTION
**Pull request by vivekmoorthy**
Add IANA-reserved port for NTP:
https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=ntp

### CODE REVIEW

**Comment by mikeberesford** [comment]
@kcudnik @rck-innovium @JaiOCP @kperumalbfn Mind reviewing this PR? It's similar to https://github.com/opencomputeproject/SAI/pull/1436

File: inc/saihostif.h [hunk 1]
```diff
@@ -413,6 +413,12 @@ typedef enum _sai_hostif_trap_type_t
      */
     SAI_HOSTIF_TRAP_TYPE_P4RT = 0x0000400b,
 
+    /**
+     * @brief NTP traffic (UDP/TCP src port == 123 or UDP/TCP dst port == 123)
+     * to local router IP address (default packet action is drop)
+     */
+    SAI_HOSTIF_TRAP_TYPE_NTP = 0x0000400c,
```
Comment (vivekmoorthy) [inline inc/saihostif.h:420]: Does it make sense to split this to NTP_CLIENT and NTP_SERVER?

**Review by kcudnik — APPROVED** [review]

**Review by JaiOCP — APPROVED** [review]
Looks good to me

File: inc/saihostif.h [hunk 2]
```diff
@@ -413,6 +413,18 @@ typedef enum _sai_hostif_trap_type_t
      */
     SAI_HOSTIF_TRAP_TYPE_P4RT = 0x0000400b,
 
+    /**
+     * @brief NTPCLIENT traffic (UDP/TCP src port == 123)
+     * to local router IP address (default packet action is drop)
+     */
+    SAI_HOSTIF_TRAP_TYPE_NTPCLIENT = 0x0000400c,
```
Comment (kperumalbfn) [inline inc/saihostif.h:420]: Do we need both TCP and UDP port 123 or just UDP 123 for NTP?

Comment (vivekmoorthy) [reply inc/saihostif.h:420]: IANA lists port 123 registered for NTP over both TCP and UDP. 
https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=ntp

**Review by kperumalbfn — APPROVED** [review]

**Comment by vivekmoorthy** [comment]
@rck-innovium Could you help with this review?

**Review by rck-innovium — APPROVED** [review]

**Comment by vivekmoorthy** [comment]
@kcudnik I see the PR is waiting on "LGTM analysis: Python"
Does this need to be triggered again? Suggestions?

**Comment by kcudnik** [comment]
not sure why this hanged on python, no changes are here made

**Comment by kcudnik** [comment]
if this retrigger will not help, please add force empty commit to this branch this will retrigger all pipelines

**Comment by kcudnik** [comment]
i'm not able to retrigger python from github page @vivekmoorthy, please add empty commit here to retrigger

**Comment by vivekmoorthy** [comment]
Thanks. retriggered and now the checks have passed.

**Review by kcudnik — APPROVED** [review]

**Comment by vivekmoorthy** [comment]
@rck-innovium @JaiOCP @kperumalbfn could you provide your approvals again due to retrigger.

**Review by kperumalbfn — APPROVED** [review]

**Comment by vivekmoorthy** [comment]
@kcudnik If approvals look ok, would you be able to help with merging this PR?

**Comment by kcudnik** [comment]
was this discussed and approved on latest SAI community meeting?

**Comment by vivekmoorthy** [comment]
> was this discussed and approved on latest SAI community meeting?

This PR was discussed 2 weeks back in SAI meeting and the decision was to split the Traps to NTPCLIENT and NTPSERVER.

@lguohan for visibility.

**Pull request merged**